packages/testing/src/test/chai-retry-plugin.unit.ts

```diff
@@ -0,0 +1,233 @@
+import Chai, { expect } from 'chai';
+import { sleep } from 'promise-assist';
+import { chaiRetryPlugin } from '../chai-retry-plugin/chai-retry-plugin';
+
+Chai.use(chaiRetryPlugin);
+
+describe('chai-retry-plugin', () => {
+    it('should retry a function that eventually succeeds', async () => {
+        const { resultFunction, getCallCount } = withCallCount((callCount: number) => {
+            if (callCount < 3) {
+                throw new Error('Failed');
+            }
+            return 'Success';
+        });
+
+        await expect(resultFunction).to.retry().to.equal('Success');
+        expect(getCallCount()).to.equal(3);
+    });
+
+    describe('options should work correctly:', () => {
+        it('timeout after the specified duration', async () => {
+            const funcToRetry = async () => {
+                await sleep(250);
+                return 'Success';
+            };
+
+            try {
+                await expect(funcToRetry).retry({ timeout: 700 }).to.equal('Failure');
+                throw new Error('This should not be called');
+            } catch (error: unknown) {
+                expect((error as Error).message).includes('Timed out after 700ms');
+            }
+        });
+
+        it('throw an error when retries limit exceeded', async () => {
+            const { resultFunction } = withCallCount((callCount) => callCount);
+
+            try {
+                await expect(resultFunction).retry({ retries: 10 }).to.be.above(100);
+                throw new Error('This should not be called');
+            } catch (error: unknown) {
+                expect((error as Error).message).to.includes('Limit of 10 retries exceeded!');
+                expect((error as Error).message).to.include('to be above 100');
+            }
+        });
+
+        it('should apply delay correctly', async () => {
+            const { resultFunction, getCallCount } = withCallCount(() => {
+                throw new Error('Im throwing');
+            });
+
+            try {
+                await expect(resultFunction).retry({ delay: 50, timeout: 500 }).to.equal(5);
+            } catch (error: unknown) {
+                expect((error as Error).message).includes('Timed out after 500ms. Error: Im throwing');
+                expect(getCallCount()).to.be.within(9, 10, `Elapsed time should be within 9-10`);
```

Comment:
why not just 9 or 10? (I guess it wi;; run 9 times). If it's too close, just tweak options a bit

Reply:
@yurii-ve there is a difference when tests are running on windows and other platforms, probably timeouts works longer there so function is called 9 times instead of 10 😅